### ISSUE TITLE
Fix xtext-maven-plugin ITs

### ISSUE THREAD

**Issue by LorenzoBettini** [xtext/xtext-monorepo#16]
While experimenting with a single Maven build, https://github.com/xtext/xtext-monorepo/issues/14 The `XtextGeneratorIT` started to show failures on Jenkins (not on GitHub Actions). The failure doesn't make sense because in the first build I didn't change anything critical.

```
21:50:51  [ERROR] Errors: 
21:50:51  [ERROR]   XtextGeneratorIT.mavenConfiguration:137->verifyErrorFreeLog:238->verifyErrorFreeLog:251 » Verification
21:50:51  [ERROR]   XtextGeneratorIT.simpleLang:88->verifyErrorFreeLog:238->verifyErrorFreeLog:251 » Verification
21:50:51  [ERROR]   XtextGeneratorIT.simpleLangIncremental:95->verifyErrorFreeLog:238->verifyErrorFreeLog:251 » Verification
21:50:51  [INFO] 
21:50:51  [ERROR] Tests run: 14, Failures: 0, Errors: 3, Skipped: 0
```

In the log, I see

```
21:50:51  [INFO] -----------------< org.eclipse.xtext:simple-lang-test >-----------------
21:50:51  [INFO] Building simple-lang-test IT-SNAPSHOT
21:50:51  [INFO] --------------------------------[ jar ]---------------------------------
21:50:51  [WARNING] The metadata /home/jenkins/.m2/repository/org/eclipse/xtext/org.eclipse.xtext.maven.parent/2.31.0-SNAPSHOT/maven-metadata-local.xml is invalid: Snapshot information corrupted with remote repository data, please verify that no remote repository uses the id 'local'
```

This message comes from https://github.com/apache/maven/blob/master/maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/DefaultVersionResolver.java#L266 and the comment says

```
 NOTE: Users occasionally misuse the id "local" for remote repos which screws up the metadata
 of the local repository. This is especially troublesome during snapshot resolution so we try
 to handle that gracefully.
```

In fact, I think we're doing the wrong thing in those integration tests.

I forgot to mention that I had already fixed problems with integration tests of the other maven plugin, xtend-maven-plugin: https://github.com/xtext/xtext-monorepo/commit/aadb4482897d943334952497d0a6bbda11a8e748

The problem is that we run the integration tests of the test projects against the same Maven local repository we use in the current build, and I guess we corrupt our local maven repository with wrong artifacts.

From what I know, the standard way to run integration tests for maven plugins is to first use the https://maven.apache.org/plugins/maven-invoker-plugin/install-mojo.html to install all the dependencies in a segregated and isolated maven repository and run the integration tests against that repository. Of course, the verifier must be set to use that local maven repository.
That's how I fixed xtend-maven-plugin https://github.com/xtext/xtext-monorepo/commit/aadb4482897d943334952497d0a6bbda11a8e748

I'll try to apply the same fix to xtext-maven-plugin

@cdietrich @szarnekow do you happen to know why we were running integration tests like that?

**Comment by LorenzoBettini** [comment]
@cdietrich @szarnekow I guess I was right and the build is back to green. If you have time, please have a look at the PR (there are also a few comments).
I'm quite confident that's the right way.

**Comment by LorenzoBettini** [comment]
Fixed by #17
